sympy/series/fourier.py

```diff
@@ -440,16 +438,210 @@ def __sub__(self, other):
+                elif q:
+                    bn[q[a]] = q[b] + bn.get(q[a], S.Zero)
+                else:
+                    a0 += p
+
+            exprs = (a0, an, bn)
+
+        args = map(sympify, (f, limits, exprs))
+
+        return Expr.__new__(cls, *args)
+
+    @property
+    def interval(self):
+        _length = 1 if self.a0 else 0
+        _length += max(set(self.an.keys()).union(set(self.bn.keys()))) + 1
+        return Interval(0, _length)
+
+    @property
+    def length(self):
+        return self.stop - self.start
+
+    @property
+    def L(self):
+        return abs(self.period[1] - self.period[0]) / 2
+
+    def truncate(self, n=oo):
+        """
+        Return the first n nonzero terms of the series if n is less than the size of the finite fourier series.
+
+        If n is None return an iterator.
+
+        See Also
+        ========
+
+        sympy.series.fourier.FourierSeries.truncate
+        """
+        if n is None:
+            return iter(self)
+
+        terms = []
+        for t in self:
+            if len(terms) == n:
+                break
+            if t is not S.Zero:
+                terms.append(t)
+
+        return Add(*terms)
 
+    def shiftx(self, s):
+        """Shift x by a term independent of x.
+
+        f(x) -> f(x + s)
 
-def fourier_series(f, limits=None):
+        This is fast, if Fourier series of f(x) is already
+        computed.
+
+        See Also
+        ========
+        sympy.series.fourier.FourierSeries.shiftx
+
+        """
+        s, x = sympify(s), self.x
+
+        if x in s.free_symbols:
+            raise ValueError("'%s' should be independent of %s" % (s, x))
+
+        _expr = self.truncate().subs(x, x + s)
+        sfunc = self.function.subs(x, x + s)
+
+        return self.func(sfunc, self.args[1], _expr)
+
+    def scale(self, s):
+        """Scale the function by a term independent of x.
+
+        f(x) -> s * f(x)
+
+        This is fast, if Fourier series of f(x) is already
+        computed.
+
+        See Also
+        ========
+        sympy.series.fourier.FourierSeries.scale
+
+        """
+        s, x = sympify(s), self.x
+
+        if x in s.free_symbols:
+            raise ValueError("'%s' should be independent of %s" % (s, x))
+
+        _expr = self.truncate() * s
+        sfunc = self.function * s
+
+        return self.func(sfunc, self.args[1], _expr)
+
+    def scalex(self, s):
+        """Scale x by a term independent of x.
+
+        f(x) -> f(s*x)
+
+        This is fast, if Fourier series of f(x) is already
+        computed.
+
+        See Also
+        ========
+        sympy.series.fourier.FourierSeries.scalex
+        """
+        s, x = sympify(s), self.x
+
+        if x in s.free_symbols:
+            raise ValueError("'%s' should be independent of %s" % (s, x))
+
+        _expr = self.truncate().subs(x, x * s)
+        sfunc = self.function.subs(x, x * s)
+
+        return self.func(sfunc, self.args[1], _expr)
```

Comment:
I would suggest to avoid redefining functions until it's necessary. Can we remove the redefined functions unless they are necessary.

Reply:
except `truncate` others I redefined have a different approach than the code in `FourierSeries`. I will reduce the redundancy with `truncate`.